libraries/chain/include/eosio/chain/block_header_state.hpp

```diff
@@ -119,6 +119,9 @@ struct block_header_state {
    digest_type compute_base_digest() const;
    digest_type compute_finality_digest() const;
 
+   // Returns true if the block is a Savanna Genesis Block
+   bool is_savanna_genesis_block() const { return core.is_genesis_block_num(block_num()); }
```

Comment:
I am not understanding how this can possibly work.  As far as I understand, we don't create a `block_header_state` or a `core` for the genesis block.

Reply:
We do create `block_header_state` and a `core` for the genesis block https://github.com/AntelopeIO/spring/blob/c7b60aad19f5b7de646c2237d3c9d79e51b41277/libraries/chain/block_state.cpp#L71

Reply:
I will add a comment to explain is_savanna_genesis_block() is only applicable for reclassified transition blocks.

Reply:
Thanks Lin, my bad, I mis-remembered!